reactioncommerce/api-plugin-tags
# fix: allow different shops to have same tag name

Signed-off-by: Mohan Narayana <mohan.narayana@mailchimp.com>

Resolves #28 
Impact: **minor**
Type: **bugfix**

## Issue
The problem is that Tag's slug is unique across the whole system. For system where there can be multiple shops it doesn't look like a good solution since it's a very common case to have same tags in different shops

## Solution
Make slug and shopId index pair value to be unique.



## Testing
1. Set up api-migrations using [these](https://github.com/reactioncommerce/api-migrations#local-development-usage) steps. NOTE: Use node 14 at step 3. Copy the new api-plugin-tags to api-migrations/node_modules/@reactioncommerce.
2. Add below code to api-migrations/migrator.config.js
    `{
      namespace: "tags",
      package: "@reactioncommerce/api-plugin-tags",
      version: 2
    }`
3. Run `MONGO_URL=mongodb://localhost:27017/reaction npx migrator migrate`
4. Check DB for index 'slug_1' in the tags table.
